src/qt/transactionrecord.cpp

```diff
@@ -49,46 +49,38 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet*
         fZSpendFromMe = wallet->IsMyZerocoinSpend(zcspend.getCoinSerialNumber());
     }
 
-    if (wtx.IsCoinStake()) {
+       if (wtx.IsCoinStake()) {
         TransactionRecord sub(hash, nTime);
         CTxDestination address;
-        if (!wtx.HasZerocoinSpendInputs() && !ExtractDestination(wtx.vout[1].scriptPubKey, address))
+        if (!ExtractDestination(wtx.vout[1].scriptPubKey, address))
             return parts;
 
-        if (wtx.HasZerocoinSpendInputs() && (fZSpendFromMe || wallet->zvitTracker->HasMintTx(hash))) {
-            //zVIT stake reward
-            sub.involvesWatchAddress = false;
-            sub.type = TransactionRecord::StakeZVIT;
-            sub.address = mapValue["zerocoinmint"];
-            sub.credit = 0;
-            for (const CTxOut& out : wtx.vout) {
-                if (out.IsZerocoinMint())
-                    sub.credit += out.nValue;
+        if (!IsMine(*wallet, address)) {
+            //if the address is not yours then it means you have a tx sent to you in someone elses coinstake tx
```

Comment:
A coinstake tx currently doesnt allow you to send the coinstake reward to another address.may wanna check how pivx detects mn rewards.

Reply:
I will test it shortly...